Add ability to run things under valgrind.

defuse opened the pull request:
If we merge this then we can add a Buildbot waterfall column. Maybe report the total number of errors to speed.z.cash as though it were a performance measure.

ebfull committed:
Enforce first four bits are zero for all spending keys and phi.

str4d committed:
Add Zcash revision to version strings

str4d committed:
Bitcoin -> Zcash in version and help text

str4d committed:
Add Zcash Developers to CLI copyright notice

defuse committed:
Add ability to run things under valgrind.

defuse committed:
Pass -DPURIFY to OpenSSL so it doesn't clutter valgrind output.

defuse committed:
Enable -v for valgrind so we can see counts for each error.

nejucomo committed:
Fix a test name bug so that ``make cov-zcash`` correctly runs the ``zcash-gtest`` binary. Fixes #946.

daira commented on qa/zcash/performance-measurements.sh:
```diff
@@ -94,6 +108,35 @@ case "$1" in
+                zcash_rpc zcbenchmark verifyequihash 1
+                ;;
+            *)
+                zcashd_massif_stop
```
Should be `zcashd_valgrind_stop`.

daira commented:
ACK, and this is good for now, but I think we will want to run valgrind on the whole test suite rather than just the benchmarks.

str4d committed:
Minor error message tweak

Commit:
Auto merge of #943 - str4d:938-clientversion, r=ebfull

Update version and help strings for Zcash

Closes #938.

loxal committed:
add sha256sum support for Mac OS X

ebfull committed:
Enable binary serializations of proofs and r1cs keys, and make the `CPourTx` proof field fixed-size.

ebfull committed:
Reorder fields of CPourTx to reflect the spec.

ebfull committed:
Update proving key and tests that depend on transaction structure changes

ebfull committed:
Refactor PRF_gadget to hand responsibility to PRF_addr_a_pk_gadget for creating the '0' argument to the PRF.

Commit:
Auto merge of #948 - nathan-at-least:t946.fix-cov-zcash, r=nathan-at-least

Fix a test name bug so that ``make cov-zcash`` correctly runs the ``zcash-gtest`` binary. Fixes #946.

alfiedotwtf committed:
Rename libzerocash to libzcash

Commit:
Auto merge of #961 - alfiedotwtf:zc.v0.11.2.latest, r=ebfull

Rename libzerocash to libzcash

Rename libzerocash to libzcash in Makefiles and source (C++ namespace) #960

Commit:
Auto merge of #958 - ebfull:prf-refactor, r=ebfull

Refactor PRF_gadget to hand responsibility to PRF_addr_a_pk_gadget for creating the '0' argument to the PRF

Closes #930

Commit:
Auto merge of #955 - loxal:zc.v0.11.2.latest, r=ebfull

#776 add sha256sum support for Mac OS X

Commit:
Auto merge of #956 - ebfull:uint252, r=ebfull

Enforce first four bits are zero for all spending keys and phi.

This introduces a `uint252` to ensure these fields are 252-bits for the PRFs.

Closes #899

Commit:
Auto merge of #954 - ebfull:fix-cpourtx-structure, r=ebfull

CPourTx structural changes

* Enable binary serialization of proofs and the proving key (closes #799) and make the proofs fixed-size.
* Reorder fields of CPourTx to match the spec (closes #927)

ebfull committed:
Enable MULTICORE proving behavior with omp.

ebfull committed:
Pass `-fopenmp` at compile-time to enable MULTICORE.
